custom_components/hafas/sensor.py

```diff
@@ -84,88 +85,56 @@ def __init__(
         self._attr_name = title
         self._attr_icon = ICON
         self._attr_unique_id = entry_id
+        self._attr_device_class = SensorDeviceClass.TIMESTAMP
         self._attr_attribution = "Provided by " + profile + " through HaFAS API"
 
         self.journeys: list[Journey] = []
 
     def calc_native_value(self) -> str:
         """Return the departure time of the next train."""
-        if (
-            len(self.journeys) == 0
-            or self.journeys[0].legs is None
-            or len(self.journeys[0].legs) == 0
-        ):
-            return "No connection possible"
+        if not self.journeys:
+            return None
 
-        first_leg = self.journeys[0].legs[0]
+        connections = to_dict(self.journeys)
+        # use get method, because an empty Journey would return {}
+        running = [x for x in connections if not x.get("canceled", True)]
 
-        value = first_leg.departure.strftime("%H:%M")
-        if (
-            first_leg.departureDelay is not None
-            and first_leg.departureDelay != timedelta()
-        ):
-            delay = int(first_leg.departureDelay.total_seconds() // 60)
-
-            value += f" + {delay}"
+        if len(running) > 0:
+            return running[0]["departure"] + (
+                dt_util.parse_duration(running[0]["delay"]) or timedelta()
```

Comment:
Should this better be max(delay, timedelta)? As a customer I'm interested in the larger of both, maybe even the sum of both.

Reply:
The `or timedelta()` part is just a fallback, in case the actual data is no timedelta, e.g. `None`.